scene/main/node.cpp

```diff
@@ -1715,8 +1715,41 @@ void Node::reparent(Node *p_parent, bool p_keep_global_transform) {
 		return;
 	}
 
+	bool preserve_owner = p_parent->is_ancestor_of(this);
```

Comment:
This is not the same as the previous code. You should check if `data.owner` is ancestor of `p_parent`.

Reply:
sheesh you're fast. you're also totally correct on that. i'll get that fixed.